src/blockscoutApi/index.ts

```diff
@@ -49,18 +49,21 @@ export class BlockscoutAPI extends DataSource {
       .catch(this.errorHandling)
   }
 
-  async getEventsByAddress (address: string) {
+  async getEventsByAddress (address: string, flow: string = 'all') {
     const params = {
       module: 'account',
       action: 'tokentx',
       address: address.toLowerCase()
     }
     return this.axios?.get<ServerResponse<TokenTransferApi>>(`${this.url}`, { params })
-      .then(response =>
-        response.data.result
-          .map(tokenTranfer => {
-            return fromApiToTEvents(tokenTranfer)
-          }))
+      .then(response => response.data.result.filter(event => {
+        if (flow === 'all') return event
+        return event[flow].toLowerCase() === address
+      }))
+      .then(response => response
+        .map(tokenTranfer => {
+          return fromApiToTEvents(tokenTranfer)
+        }))
```

Comment:
this could be rewritten as:

```
.then(response => response.map(fromApiToTEvents))
```

Since the value is coming from one function and being directly used in another.

Reply:
Done!